frame/pablo/src/lib.rs

```diff
@@ -467,6 +515,83 @@ pub mod pallet {
 		pub(crate) fn account_id(pool_id: &T::PoolId) -> T::AccountId {
 			T::PalletId::get().into_sub_account(pool_id)
 		}
+		pub(crate) fn do_get_exchange_rate(
+			pool_id: T::PoolId,
+			swap: bool,
+		) -> Result<Rate, DispatchError> {
+			let pair = Self::currency_pair(pool_id)?;
+			let pool_account = Self::account_id(&pool_id);
+			let pair = if swap { pair.swap() } else { pair };
+			let pool_base_asset_under_management =
+				T::Convert::convert(T::Assets::balance(pair.base, &pool_account));
+			let pool_quote_asset_under_management =
+				T::Convert::convert(T::Assets::balance(pair.quote, &pool_account));
+			Ok(Rate::checked_from_rational(
+				pool_base_asset_under_management,
+				pool_quote_asset_under_management,
+			)
+			.ok_or(ArithmeticError::Overflow)?)
+		}
+
+		fn update_twap(pool_id: T::PoolId) -> Result<(), DispatchError> {
+			if TWAPState::<T>::contains_key(pool_id) {
```

Comment:
Please replace `TWAPState::<T>::contains` and `TWAPState::<T>::get(pool_id).map_or(..` by doing a direct `if let Some(..) = TWAPState...get(..)`